WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/document/DocumentCreator.java

```diff
@@ -51,16 +55,22 @@ private static String getEditorOrFileContent(ITextEditor editor) {
 		} catch (IllegalArgumentException | ContentReaderException exception) {
 			// Editor was null, there is nothing we can do to get the file
 			// contents.
-			WatchDogLogger.getInstance().logSevere(exception);
+			WatchDogLogger.getInstance(

```

Comment:
The logic is the following:
- flag to enable logging is in Preferences (plugin-specific);
- logger is moved to the core, so from getInstance it cannot access the current Preferences, and see should it be enabled or not;
-whoever calls a Logger now tells it whether is it enabled or not.

However, I think I have a mechanism to avoid this, with Preferences kept in WatchDogGlobals,  so I will fix this.